test/native/basic_test.dart

```diff
@@ -305,10 +305,11 @@ void main() {
       // 4. Now second connection is ready. Second query has two connections to choose from.
       // 5. However, first connection is closed, so it's removed from the pool.
       // 6. Triggers `Concurrent modification during iteration: Instance(length:1) of '_GrowableList'`
-      final db = await testUtils.setupDatabase(path: path, initStatements: [
+      final db = SqliteDatabase.withFactory(
```

Comment:
why is this change needed?

Reply:
[Discussed offline]  This is inline with how `master` currently is.